### PR TITLE
Update xchain-thorchain (prepare bump @0.6.1)

### PR DESCRIPTION
**Pull request by junkai121**
### Update

- Add `setClientUrl`
- Add `getDefaultClientUrl`

### Fix

- Fix client URL for multichain testnet (`https://testnet.thornode.thorchain.info`)

### CODE REVIEW

File: packages/xchain-thorchain/src/client.ts [hunk 1]
```diff
@@ -68,14 +68,12 @@ class Client implements ThorchainClient, XChainClient {
...
-    this.thorClient = new CosmosSDKClient({
-      server: this.getClientUrl(),
-      chainId: this.getChainId(),
-      prefix: this.getPrefix(),
-      derive_path: this.derive_path,
-    })
-    this.address = ''
+    if (this.network != network) {
+      this.network = network
+      this.clientUrl = this.getDefaultClientUrl()
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:73]: ^ That would deprecate/override `clientUrl`, which may set using `setClientUrl` before.

File: packages/xchain-thorchain/src/client.ts [hunk 2]
```diff
@@ -84,8 +82,26 @@ class Client implements ThorchainClient, XChainClient {
     return this.network
   }
 
+  setClientUrl = (clientUrl: string): void => {
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:85]: How about: `clientUrl: string` -> `{testnetClientUrl: string, mainnetClientUrl: string}`

File: packages/xchain-thorchain/src/client.ts [hunk 3]
```diff
@@ -51,12 +55,8 @@ class Client implements ThorchainClient, XChainClient {
 
   constructor({ network = 'testnet', phrase }: XChainClientParams) {
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:56]: How about to provide `clienUrl` for `test|mainnet` as optional parameters? 
- `testnetClientUrl: string`
- `mainnetClientUrl: string`

File: packages/xchain-thorchain/src/client.ts [hunk 4]
```diff
@@ -84,8 +82,26 @@ class Client implements ThorchainClient, XChainClient {
...
+    this.clientUrl = clientUrl
+    this.thorClient = this.getNewThorClient()
+  }
+
   getClientUrl = (): string => {
-    return this.network === 'testnet' ? 'http://18.198.92.45:1317' : 'http://138.68.125.107:1317'
+    return this.clientUrl
+  }
+
+  getDefaultClientUrl = (): string => {
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:94]: In addition to previous comment, return of `{testnetClientUrl: string, mainnetClientUrl: string}` is needed then. 

And you may need to have another function `getClientUrlByNetwork = (network: Network): string`.

File: packages/xchain-thorchain/src/client.ts [hunk 5]
```diff
@@ -84,8 +82,26 @@ class Client implements ThorchainClient, XChainClient {
     return this.network
   }
 
+  setClientUrl = (clientUrl: string): void => {
+    this.clientUrl = clientUrl
+    this.thorClient = this.getNewThorClient()
+  }
+
   getClientUrl = (): string => {
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:90]: In addition to previous comment ^
`getClientUrl = (): {testnetClientUrl: string, mainnetClientUrl: string}`

**Review by veado — CHANGES_REQUESTED** [review]

File: packages/xchain-thorchain/src/client.ts (hunk 5 above)
Comment (junkai121) [reply packages/xchain-thorchain/src/client.ts:90]: What do you think of following approach? @veado  It returns the client url of current network...
```
  getClientUrl = (): string => {
    return this.getClientUrlByNetwork(this.network)
  }
```

**Review by veado — APPROVED** [review]

**Pull request merged**